tests/FunctionAllTest.php

```diff
@@ -94,4 +94,25 @@ public function shouldResolveToEmptyArrayWhenInputPromiseDoesNotResolveToArray()
         all(resolve(1))
             ->then($mock);
     }
+
+    /** @test */
+    public function shouldPreserveTheOrderOfArrayWhenResolvingAsyncPromises()
+    {
+        $queue = new \SplQueue();
+        $asyncPromise = new Promise(function ($resolve) use ($queue) {
+            $queue->enqueue(function () use ($resolve, $queue) {
+                $resolve(2);
+            });
+        });
+
+        $result = null;
+
+        all([resolve(1), $asyncPromise, resolve(3)])->then(function ($resolvedValues) use (&$result) {
+            $result = $resolvedValues;
+        });
+
+        call_user_func($queue->dequeue());
+
+        $this->assertSame([1, 2, 3], $result);
+    }
```

Comment:
I'd like to see the style of the test matching the rest of the tests. In this case this would something like

```php
/** @test */
public function shouldPreserveTheOrderOfArrayWhenResolvingAsyncPromises()
{
    $deferred = new Deferred();

    $mock = $this->createCallableMock();
    $mock
        ->expects($this->once())
        ->method('__invoke')
        ->with($this->identicalTo([1, 2, 3]));

    all([1, $deferred->promise(), 3])
        ->then($mock);

    $deferred->resolve(2);
}
```